tests/rptest/tests/partition_force_reconfiguration_test.py

```diff
@@ -368,25 +370,95 @@ def get_lso():
         self.start_consumer()
         self.run_validation()
 
-    @cluster(num_nodes=5)
+
+class NodeWiseRecoveryTest(RedpandaTest):
+    def __init__(self, test_context, *args, **kwargs):
+        super(NodeWiseRecoveryTest,
+              self).__init__(test_context,
+                             si_settings=SISettings(
+                                 log_segment_size=1024 * 1024,
+                                 test_context=test_context,
+                                 fast_uploads=True,
+                                 retention_local_strict=True,
+                             ),
+                             extra_rp_conf={
+                                 "partition_autobalancing_mode": "continuous",
+                                 "enable_leader_balancer": False,
+                             },
+                             num_brokers=5,
+                             *args,
+                             **kwargs)
+        self.default_timeout_sec = 60
+        self.rpk = RpkTool(self.redpanda)
+        self.admin = Admin(self.redpanda)
+
+    def _alive_nodes(self):
+        return [n.account.hostname for n in self.redpanda.started_nodes()]
+
+    def collect_topic_partition_states(self, topic):
+        states = []
+        for p in self.rpk.describe_topic(topic):
+            states.append(
+                self.admin.get_partition_state(
+                    namespace="kafka",
+                    topic=topic,
+                    partition=p.id,
+                    node=self.redpanda.get_node_by_id(p.leader)))
+        return states
+
+    def get_topic_partition_high_watermarks(self, topic):
+        statuses = self.collect_topic_partition_states(topic)
+        return [[r['high_watermark'] for r in s['replicas']] for s in statuses]
+
+    def producer_until_segments_removed(self, topic):
+        msg_size = 512
+
+        self.producer = KgoVerifierProducer(self.test_context, self.redpanda,
+                                            topic, msg_size, 10000000)
+
+        self.producer.start(clean=False)
+
+        def all_cloud_offsets_advanced():
```

Comment:
simplify as `return all(r['next_cloud_offset'] >= 1000 for s in states for r in s['replicas'])` ?